test/functional/test_framework/messages.py

```diff
@@ -1045,13 +1045,11 @@ def deserialize(self, f):
 
         self.nStartingHeight = struct.unpack("<i", f.read(4))[0]
 
-        if self.nVersion >= 70001:
-            # Relay field is optional for version 70001 onwards
-            try:
-                self.relay = struct.unpack("<b", f.read(1))[0]
-            except:
-                self.relay = 0
-        else:
+        # Relay field is optional for version 70001 onwards
+        # But, unconditionally check it to match behaviour in bitcoind
+        try:
```

Comment:
The goal of the test framework is not to match behavior in bitcoind (we don't want to re-implement bitcoind in python), but to properly deserialize messages that have been serialized by bitcoind.

Thus, the try-catch can be removed completely.

Reply:
The deserialization code can be useful in other contexts, eg https://github.com/bitcoin/bitcoin/blob/f67b5dca576669bfed0cf4af9f2cc6273010a0d7/contrib/message-capture/message-capture-parser.py. Besides, there's no disadvantage to making this more robust.